Dalamud/Game/ClientState/Actors/Types/Actor.cs

```diff
@@ -71,6 +73,86 @@ public class Actor : IEquatable<Actor> {
         /// </summary>
         public virtual int TargetActorID => 0;
 
+        public ObjectVisibility ObjectVisibility => this.actorStruct.ObjectVisibility;
+
         bool IEquatable<Actor>.Equals(Actor other) => this.ActorId == other.ActorId;
+
+        /// <summary>
+        /// Hides the actor
+        /// </summary>
+        public async void Hide()
+        {
+            var player = ObjectKind == ObjectKind.Player;
+
+            unsafe
+            {
+                ObjectKind* kind = (ObjectKind*)(Address + ActorOffsets.ObjectKind);
```

Comment:
use `var`, there's no point in specifying the type on the left hand side when it's given on the right

Reply:
and again mark func as unsafe and do this all once

Reply:
can't do unsafe func + ``await Task.Delay(100)``

Reply:
lol async is such garbage, a Task.Run then with system.threading.sleep would probably make it much nicer then

Reply:
Thread sleep stops all tasks from running through, so its not ideal

Reply:
shouldn't do because tasks are pooled?

Reply:
I'm likely to be misinterpreting something, but there was issues with it in a plugin previously I think
https://github.com/attickdoor/XIVComboPlugin/commit/75b9c8e4

Reply:
I just tried the Thread.Sleep inside of Task.Run it and it doesn't seem to lock up my game

Reply:
test with sonar enabled as well, that was the plugin causing issues

Reply:
If its working fine its fine, Adam usually knows what hes saying so probably safe to go with his suggestions lol

Reply:
a regular thread would work I guess, which wouldn't cause any issues of the sort but this is probably one of those things where it'd be better to use a scheduler instead

no idea why sonar would hang the game if a task is blocking though, must be doing some real stupid shit for that to happen given that tasks wouldn't be executing on a game thread, unless they're awaiting on a game thread which is lol tier

Reply:
yea works fine with Task.Run, I haven't tested it Sonar yet since I don't really want to run that plugin on my main computer (and it's possible that their plugin is doing something wrong, forcing everything to adjust to that isn't really a good thing I think...)

Reply:
yeah if it's doing something stupid like awaiting tasks on a game thread, then that's sonars problem and nobody elses because that absolutely will block and cause the game to hitch